code/common/checkinputs.q

```diff
@@ -0,0 +1,276 @@
+    :checkeachpair[raze each flip xinvalidpairs];
+  };
+
+checkeachpair:{[invalidpair]'`$.checkinputs.formatstring["Parameter:{parameter} cannot be used in conjunction with parameter(s):{invalidpairs}";invalidpair]};
+
+// function to check if any parameters are repeated
+checkrepeatparams:{[dict]
+    if[any repeats:1<count each group key dict;
+        '`$.checkinputs.formatstring["{} parameter(s) used more than once";where repeats]];
+    :dict;};
+
+// loop thorugh input parameters to execute parameter specific checks
+checkeachparam:{[dict;isrequired]
+    config:select from .checkinputs.checkinputsconfig where parameter in key dict,required=isrequired;
+    :checkparam/[dict;config];
+  };
+
+// extract parameter specific function from config to check the input
+checkparam:{[dict;config]
+    (first config[`checkfunction])[dict;first config`parameter]};
+
+// check tablename parameter is of type symbol
+checktable:{[dict;parameter]:checktype[-11h;dict;parameter];};
+
+// check that endtime is of temporal type and that it is greater than or equal to starttime
+checkendtime:{[dict;parameter]
+    dict:checktimetype[dict;parameter];
+    :checktimeorder dict};
+
+// check that inputted value is of valid type: -12 -14 -15h
+checktimetype:{[dict;parameter]:checktype[-12 -14 -15h;dict;parameter];};
+
+// check timecolumn is of type symbol
+// check starttime <= endtime
+checktimecolumn:{[dict;parameter]:checktype[-11h;dict;parameter];};
+
+// check starttime <= endtime
+checktimeorder:{[dict]
+    if[dict[`starttime] > dict`endtime;'`$"Starttime parameter must be <= endtime parameter"];
+    :dict;};
+
+// check instruments are of type symbol
+checkinstruments:{[dict;parameter]
+    :checktype[-11 11h;dict;parameter];};
+
+// check columns are of type symbol
+checkcolumns:{[dict;parameter]
+    :checktype[-11 11h;dict;parameter];};
+
+// check groupings are of type symbol
+checkgrouping:{[dict;parameter]
+    :checktype[-11 11h;dict;parameter];};
+
+// check aggregations are of type dictionary, that the dictionary has symbol keys, that 
+// the dictionary has symbol values
+checkaggregations:{[dict;parameter]
+    example:"`max`min`wavg!(`price;`size;`price`size)";
+    dict:checktype[99h;dict;parameter];
+    input:dict parameter;
+    if[not 11h~abs type key input;
+        '`$"Aggregations parameter key must be of type 11h - example: ",example];
+    if[not all 11h~/:abs raze type''[get input];
+        '`$"Aggregations parameter values must be of type symbol - example: ",example];
+    :dict;
+  };
+
+// check that timebar parameter has three elements of respective types: numeric, symbol,
+// symbol.
+checktimebar:{[dict;parameter]
+    input:dict parameter;
+    if[not(3=count input)&0h~type input;
+        '`$"Timebar parameter must be a list of length 3"];
+    input:`size`bucket`timecol!input;
+    if[not any -6 -7h~\:type input`size;
+        '`$"First argument of timebar must be either -6h or -7h"];
+    if[not -11h~type input`bucket;
```

Comment:
checktype takes in three arguments; valid types, the query as a dictionary and the parameter to be checked. An error message is generated from these arguments. The timebar parameter is a mixed list of one number and two symbols and checktimebar checks each element in timebar is of the correct type. checktype can only check if timebar is type 0h

Reply:
Ok - did you manage to confirm if it could be tested using checktype instead?

Reply:
We can if we define a new dictionary using the timebar parameter, the only catch is that it would display the error message as first_argument_of_timebar input type incorrect - valid type(s):-6h -7h - input type:-11h